target_chains/cosmwasm/tools/src/chains-manager/injective.ts

```diff
@@ -0,0 +1,231 @@
+    }
+
+    return txResponse;
+  }
+
+  async storeCode(req: StoreCodeRequest): Promise<StoreCodeResponse> {
+    const { contractBytes } = req;
+    const store_code = MsgStoreCode.fromJSON({
+      sender: this.getAddress(),
+      wasmBytes: contractBytes,
+    });
+
+    const txResponse = await this.signAndBroadcastMsg(store_code);
+
+    const codeId: number = parseInt(
+      extractFromRawLog(txResponse.rawLog, "code_id")
+    );
+
+    return {
+      txHash: txResponse.txHash,
+      codeId,
+    };
+  }
+
+  async instantiateContract(
+    req: InstantiateContractRequest
+  ): Promise<InstantiateContractResponse> {
+    const { codeId, instMsg, label } = req;
+    const instantiateMsg = MsgInstantiateContract.fromJSON({
+      sender: this.getAddress(),
+      admin: this.getAddress(),
+      codeId,
+      label,
+      msg: instMsg,
+    });
+
+    const txResponse = await this.signAndBroadcastMsg(instantiateMsg);
+
+    const contractAddr: string = extractFromRawLog(
+      txResponse.rawLog,
+      "contract_address"
+    );
+
+    return {
+      txHash: txResponse.txHash,
+      contractAddr,
+    };
+  }
+
+  async executeContract(
+    req: ExecuteContractRequest
+  ): Promise<ExecuteContractResponse> {
+    const { contractAddr, msg, funds } = req;
+
+    const executeMsg = MsgExecuteContract.fromJSON({
+      sender: this.getAddress(),
+      contractAddress: contractAddr,
+      msg,
+      funds,
+    });
+
+    const txResponse = await this.signAndBroadcastMsg(executeMsg);
+
+    return {
+      txHash: txResponse.txHash,
+    };
+  }
+
+  async migrateContract(
+    req: MigrateContractRequest
+  ): Promise<MigrateContractResponse> {
+    const { newCodeId, contractAddr, migrateMsg } = req;
+    const migrate_msg = MsgMigrateContract.fromJSON({
+      sender: this.getAddress(),
+      contract: contractAddr,
+      codeId: newCodeId,
+      msg: migrateMsg ?? {
+        action: "",
+      },
+    });
+
+    const txResponse = await this.signAndBroadcastMsg(migrate_msg);
+
+    let resultCodeId: number = parseInt(
+      extractFromRawLog(txResponse.rawLog, "code_id")
+    );
+    try {
+      assert.strictEqual(newCodeId, resultCodeId);
+    } catch (e) {
+      console.error("The resultant code id doesn't match newCodeId");
+      console.error(txResponse.rawLog);
+      throw e;
+    }
+
+    return {
+      txHash: txResponse.txHash,
+    };
+  }
+
+  async updateContractAdmin(
+    req: UpdateContractAdminRequest
+  ): Promise<UpdateContractAdminResponse> {
+    const { newAdmin, contractAddr } = req;
+    const currAdmin = this.getAddress();
+
+    const updateAdminMsg = new MsgUpdateAdmin({
+      sender: currAdmin,
+      newAdmin,
+      contract: contractAddr,
+    });
+
+    const txResponse = await this.signAndBroadcastMsg(updateAdminMsg);
+
+    return {
+      txHash: txResponse.txHash,
+    };
+  }
+}
+
+// enter key of what to extract
+function extractFromRawLog(rawLog: string, key: string): string {
```

Comment:
this is function exists twice in this PR (and also exists in the pusher code i presume). I think you should reuse a single implementation in this PR at least, and then we should think about deduplicating across the pusher as well.

Reply:
it doesn't exist in price pusher. 

In this pr, this was a bug. both implementations are different at a very small point.

Reply:
ooh they are different... ok nevermind. 

(I think you could extract out a helper function with some of the shared logic (and just call that helper with a different regexp here), but not a big deal)